src/test/regress/expected/full_join.out

```diff
@@ -88,6 +90,32 @@ SELECT * FROM test_table_1 FULL JOIN test_table_3 USING(id, val1) ORDER BY 1;
   4 |    5
 (4 rows)
 
+-- Full join with complicated target lists
+SELECT count(DISTINCT id), (avg(test_table_1.val1) + id * id)::integer as avg_value, id::numeric IS NOT NULL as not_null 
+FROM test_table_1 FULL JOIN test_table_3 using(id)
+WHERE id::bigint < 55
+GROUP BY id
+ORDER BY 2
+ASC LIMIT 3;
+ count | avg_value | not_null 
+-------+-----------+----------
+     1 |         2 | t
+     1 |         6 | t
+     1 |        12 | t
+(3 rows)
+
+SELECT max(val1)
```

Comment:
Note that the issue I brought up happens on non-full outer joins such as left join. Does this commit consider that?

Reply:
Yes it does, test with left join is added as well.